src/Database/Database.php

```diff
@@ -5185,7 +5185,11 @@ public function findOne(string $collection, array $queries = []): false|Document
 
         $this->trigger(self::EVENT_DOCUMENT_FIND, $found);
 
-        return $found;
+        if (!$found) {
+            return new Document();
+        } else {
+            return $found;
+        }
```

Comment:
Stylistically and for readability it's a bit better to avoid else if we can:

```suggestion
        if (!$found) {
            return new Document();
        }
        
        return $found;
```